docs/developer/payments.mdx

```diff
@@ -0,0 +1,524 @@
+[TRANSACTION_CANCELATION_REQUESTED](../api-reference/enums/webhook-event-type-sync-enum#code-style-fontweight-normal-webhookeventtypesyncenumbtransaction_cancelation_requestedbcode),
+[TRANSACTION_REFUND_REQUESTED](../api-reference/enums/webhook-event-type-sync-enum#code-style-fontweight-normal-webhookeventtypesyncenumbtransaction_refund_requestedbcode).
+
+The below diagram shows an example of processing synchronous refund action.
+
+```mermaid
+sequenceDiagram
+    Staff ->>+ Saleor: Request refund<br>[transactionActionRequest]
+    Saleor ->> Saleor: Create REQUEST event for transaction
+    Saleor ->> Saleor: Schedule transaction action webhook
+    Saleor -->>- Staff: Requested
+    Note over Saleor,App: Celery task
+    Saleor ->>+ App: Send transaction-refund-requested <br>webhook
+    App ->>+ PaymentProvider: Request refund
+    PaymentProvider -->>- App: Refunded
+    App -->>- Saleor: Return PSP<br> Reference and details<br> about completed action
+    Saleor ->> Saleor: Update transaction amounts
+```
+
+#### Saleor 3.4 (Deprecated)
+
+If your app should also process payment request actions (like charge, refund, or CANCEL) triggered by staff users on the Saleor dashboard side or by the app,
+make sure that your app subscribes to the [transaction-action-request](developer/extending/apps/sample-webhook-payloads.mdx#transaction-action-request) webhook.
+
+The chart below shows a workflow for handling a `refund` request.
+
+```mermaid
+sequenceDiagram
+    Staff->>+Dashboard: Click refund button<br> for single transaction
+    Dashboard->>+Saleor: Call transactionActionRequest<br> with action - refund
+    Saleor -->>- Dashboard: OK
+    Dashboard -->>- Staff: Triggered
+    Saleor ->>+ App: Send webhook<br>transaction-action-request<br> with all details related<br> to action and transaction
+    App -->>- Saleor: OK
+    App ->>+ PaymentProvider: Execute refund <br>action
+    PaymentProvider -->>- App: Refunded
+    App ->>+ Saleor: Call transactionUpdate <br>to update transaction with<br> a result received after <br>requested action
+    Saleor -->>- App: Updated transaction
+```
+
+### Re-calculations of transaction amounts
+
+The re-calculation will differ based on the value of the [TransactionEventTypeEnum](../api-reference/enums/transaction-event-type-enum) provided as the field `type`
+used in the [transactionEventReport](#reporting-action-for-transaction) mutation and [optionally in the webhook response](extending/apps/synchronous-webhooks/transaction-webhooks):
+
+#### AUTHORIZATION_SUCCESS
+
+Saleor assigns provided `amount` to `transaction.authorizedAmount`. You can’t have two successful authorizations for a single transaction.
+Each successful request after the first one will raise an exception. `AUTHORIZATION_ADJUSTMENT` should be used to change the `transaction.authorizedAmount`.
```

Comment:
I like that we have the description of what each event does to the amounts. However it's still a bit hard to comprehend what happens to the transaction amounts e.g. when you have 1) AUTHORIZATION_SUCCESS 2) AUTHORIZATION_FAILURE 3) AUTHORIZATION_SUCCESS

Maybe we should add a table with events and what the resulted amount is?

```
+-----------------------+--------+---------------+------------------------------+
| Event                 | Amount | PSP Reference | Transaction.authorizedAmount |
+-----------------------+--------+---------------+------------------------------+
| AUTHORIZATION_SUCCESS | 10     | 1234          | 10                           |
+-----------------------+--------+---------------+------------------------------+
| AUTHORIZATION_FAILURE |        | 1234          | 0                            |
+-----------------------+--------+---------------+------------------------------+
| AUTHORIZATION_SUCCESS | 10     | 1234          | 10                           |
+-----------------------+--------+---------------+------------------------------+
```

Reply:
I will extend the re-calculation chapter, but I will do that in a separate PR as I am still going to provide the docs for support-for-payment-apps part.